src/Components/Display/Menu/UpMenu.tsx

```diff
@@ -36,25 +36,32 @@ class UpMenu extends React.Component<UpMenuProps & WithThemeProps, UpMenuState>{
 
     static defaultProps = {
         theme: defaultTheme,
-        icon: (props) => <UpSvgIcon width={48} height={48} iconHtml={logo}></UpSvgIcon>,
+        icon: () => <UpSvgIcon width={48} height={48} iconHtml={logo}></UpSvgIcon>,
         width: '275px',
-        minified:false
     }
 
     constructor(p, c) {
         super(p, c);
         this.state = {
-            minified: this.props.minified,
+            minified: false,
         };
     }
 
     toggleMinification = () => {
-        this.setState({ minified: !this.state.minified },()=>{
+        this.setState(!this.isMinifiedControlled?{ minified: !this.currentMinifiedValue }:null,()=>{
             if(this.props.onMinifiedChange){
-                this.props.onMinifiedChange(this.state.minified);
+                this.props.onMinifiedChange(this.currentMinifiedValue);
```

Comment:
Dans le cas où la valeur est controllée tu passeras toujours la même valeur au handler onMinifiedChange.
if(this.isMinifiedControlled && this.props.onMinifiedChange) {
this.props.onMinifiedChange(!this.currentMinifiedValue)
} else {
this.setState({ minified: !this.currentMinifiedValue },()=>{
            if(this.props.onMinifiedChange){
                this.props.onMinifiedChange(this.currentMinifiedValue);
}
}

Reply:
j'ai apporté les modifications